### PR TITLE
Option to specify cmd line params for caged programs

### PR DESCRIPTION
**Pull request by DonatJR**
image

### CODE REVIEW

File: SharkCage/SharedFunctionality/SharedFunctions.cpp [hunk 1]
```diff
@@ -134,6 +136,14 @@ namespace SharedFunctions
 				config_stream >> json_config;
 
 				auto path = json_config[APPLICATION_PATH_PROPERTY].get<std::string>();
+				
+				auto version = json_config[CONFIG_VERSION_PROPERTY].get<int>();
+				
+				std::string cmd_line = "";
+				if (version >= 2)
+				{
+					cmd_line = json_config[APPLICATION_CMD_LINE_PROPERTY].get<std::string>();
```
Comment (SailReal) [inline SharkCage/SharedFunctionality/SharedFunctions.cpp:145]: `""` could be replaced with `String.Empty` or?

File: SharkCage/SharedFunctionality/CageData.h [hunk 2]
```diff
@@ -7,6 +7,7 @@ struct CageData
 {
 	const std::wstring config_path;
 	std::wstring app_path;
+	std::wstring app_cmd_line;
```
Comment (SailReal) [inline SharkCage/SharedFunctionality/CageData.h]: Maybe rename `app_cmd_line` to `app_cmd_line_params`?

**Review by SailReal — CHANGES_REQUESTED** [review]
Looks awesome and works like expected with Firefox 🎉 .

Maybe we can add a tool tip in the Configurator which describes for what the params are used and maybe with an example e.g.`www.wikipedia.org`

File: SharkCage/SharedFunctionality/SharedFunctions.cpp (hunk 1 above)
Comment (DonatJR) [reply SharkCage/SharedFunctionality/SharedFunctions.cpp:145]: not in C++, no 😅

**Comment by DonatJR** [comment]
instead of the new text box for the command line, there is now an advanced config where we can add more options later on:
advanced

**Review by SailReal — APPROVED** [review]

**Comment by SailReal** [comment]
Looks awesome now 👍 thanks for the enhancement 🎉, will create a 1.1.0 for that 😄

**Pull request merged**